app/utils.py

```diff
@@ -0,0 +1,25 @@
+from datetime import datetime
+from functools import partial
+from calendar import timegm
+from time import strptime
+from settings import FORMATS
+
+
+def check_datetime_format(format, date: str):
+    try:
+        return bool(datetime.strptime(date, format))
+    except ValueError:
+        return False
+
+
+def convert_time_range(time_range: list):
+    try:
+        new_time_range = []
+        for time_ in time_range:
+            index = list(
+                map(partial(check_datetime_format, date=time_), FORMATS)
+            ).index(True)
+            new_time_range.append(timegm(strptime(time_, FORMATS[index])) * 1000)
+        return new_time_range
+    except Exception:
+        return None, None
```

Comment:
Will it be a better choice to leave the caller with choice of what to do with badly formatted inputs, instead of returning `None, None`?

Reply:
In this case user wouldn't know what he did wrong

Reply:
How the user will not what he did wrong when function returns the same result `None, None` for all type of errors?

Reply:
In this particular case it's impossible but if specify the error then user will understand. I agree with your first comment